rider-godot/src/main/kotlin/com/jetbrains/rider/plugins/godot/run/configurations/GodotDebugFactory.kt

```diff
@@ -0,0 +1,19 @@
+import com.intellij.execution.configurations.ConfigurationType
+import com.intellij.openapi.project.Project
+import com.jetbrains.rider.plugins.godot.GodotIcons
+import com.jetbrains.rider.plugins.godot.run.configurations.GodotDebugRunConfiguration
+import com.jetbrains.rider.run.configurations.DotNetConfigurationFactoryBase
+
+open class GodotDebugFactory(type: ConfigurationType)
+    : DotNetConfigurationFactoryBase<GodotDebugRunConfiguration>(type) {
+
+    override fun createTemplateConfiguration(project: Project) = GodotDebugRunConfiguration(project, this)
+    override fun isConfigurationSingletonByDefault() = true
+    override fun getName() = "Godot Debug"
+    override fun getIcon() = GodotIcons.RunConfigurations.StartAndDebug
+
+    // This value gets written to the config file. By default it defers to getName, which is what happened pre-2018.3.
+    // Keep the "GodotDebug" value so that we can load configs created by earlier versions, and earlier versions can
+    // load this config
```

Comment:
If this is a new configuration targeting only Rider 2019.3 (or even 2020.1?), then you shouldn't care about these details. This comment isn't very meaningful for IntelliJ 2020.1, since you don't need to keep compatibility with any earlier IDE versions.